metrics_layer/cli/seeding.py

```diff
@@ -304,12 +328,71 @@ def make_fields(self, column_data: str):
                 field["type"] = metrics_layer_type
             fields.append(field)
+        if searchable_field_candidates:
+            column_cardinalities_query = self.column_cardinalities_query(
+                column_names=searchable_field_candidates, schema_name=schema_name, table_name=table_name
+            )
+            column_cardinalities = self.run_query(query=column_cardinalities_query)
+
+            # Get the column names that have a cardinality of less than 100
+            # Note: running the query doesn't preserve the column name cases
+            searchable_column_names = [
+                col.lower().rsplit("_cardinality", 1)[0]
+                for col in column_cardinalities.columns
+                if column_cardinalities.loc[0, col] < 100
+            ]
+
+            for field in fields:
+                if field["sql"].split(".", 1)[1].lower() in searchable_column_names:
+                    field["searchable"] = True
+
         return fields
 
+    def column_cardinalities_query(self, column_names: List[str], schema_name: str, table_name: str) -> str:
+        cardinality_queries = []
+        for column_name in column_names:
+            if self.connection.type in (Definitions.snowflake, Definitions.duck_db, Definitions.druid):
+                query = (
+                    f'APPROX_COUNT_DISTINCT( "{column_name}" ) as "{column_name}_cardinality"'  # noqa: E501
+                )
+            elif self.connection.type == Definitions.redshift:
+                query = f'APPROXIMATE COUNT(DISTINCT "{column_name}" ) as "{column_name}_cardinality"'  # noqa: E501
+            elif self.connection.type == Definitions.postgres:
+                query = f'COUNT(DISTINCT "{column_name}" ) as "{column_name}_cardinality"'  # noqa: E501
+            elif self.connection.type == Definitions.sql_server:
+                query = (
+                    f'APPROX_COUNT_DISTINCT( "{column_name}" ) as "{column_name}_cardinality"'  # noqa: E501
+                )
+            elif self.connection.type == Definitions.bigquery:
+                query = f"COUNT(DISTINCT `{column_name}` ) as `{column_name}_cardinality`"
+            else:
+                raise NotImplementedError(f"Unknown connection type: {self.connection.type}")
+            cardinality_queries.append(query)
+
+        query = f"SELECT {', '.join(cardinality_queries)}"
+
+        if self.connection.type in {
+            Definitions.snowflake,
+            Definitions.duck_db,
+            Definitions.druid,
+            Definitions.redshift,
+            Definitions.postgres,
+            Definitions.sql_server,
+        }:
+            query += f" FROM {self.database}.{schema_name}.{table_name}"
+        elif self.connection.type == Definitions.bigquery:
+            query += f" FROM `{self.database}`.`{schema_name}`.`{table_name}`"
+
+        return query + ";" if self.connection.type != Definitions.druid else query
+
```

Comment:
This function implements the column cardinality queries.

It takes in a list of column names, constructs the COUNT DISTINCT queries for each column name (names this as {column_name}_cardinality). Then joins all the COUNT_DISTINCT queries into a single query before running it. 

Quotations are added around the column names to handle SQL wildcard characters (e.g. %). BigQuery uses backticks.

Reply:
This looks good to me, can you confirm you've confirmed the syntax for the db flavors with their docs and tested snowflake end to end?